### PR TITLE
Chat Removal

### PR DESCRIPTION
**Pull request by CommanderStorm**
## Issue
removed most of the chat and its dependencys

This fixes the following issue(s):
- Resolves: #1455
- Resolves: #1277

## Screenshot
image

## Why this is useful for all students
The chat is partially buggy, not state of the art and the general usefullness for studnents is at question

### CODE REVIEW

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/onboarding/OnboardingExtrasFragment.kt [hunk 1]
```diff
@@ -100,69 +87,38 @@ class OnboardingExtrasFragment : FragmentForLoadingInBackground<ChatMember>(
...
+            return false
         }
 
         // Try to restore already joined chat rooms from server
-        return try {
-            val verification = TUMCabeVerification.create(requireContext(), null)
-            val rooms = tumCabeClient.getMemberRooms(member.id, verification)
-            chatRoomController.replaceIntoRooms(rooms)
+        try {
+            TUMCabeVerification.create(requireContext(), null) // TODO is this needed?
```
Comment (CommanderStorm) [inline app/src/main/java/de/tum/in/tumcampusapp/component/ui/onboarding/OnboardingExtrasFragment.kt:105]: I dont fully know if this is needed

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/EventsViewModel.kt [hunk 2]
```diff
@@ -19,6 +19,7 @@ import io.reactivex.schedulers.Schedulers
 import java.util.concurrent.TimeUnit
 import javax.inject.Inject
 
+// TODO: is this class nessesary?
```
Comment (CommanderStorm) [inline app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/EventsViewModel.kt:22]: I have no idea, if this is the best solution..

File: app/src/main/java/de/tum/in/tumcampusapp/service/FcmReceiverService.kt [hunk 3]
```diff
@@ -30,7 +28,7 @@ class FcmReceiverService : FirebaseMessagingService() {
 
         // Legacy messages need to be handled - maybe some data is missing?
         if (!data.containsKey(PAYLOAD) || !data.containsKey("type")) {
-            handleLegacyNotification(data)
+            // TODO: is this path nessesary?????
```
Comment (CommanderStorm) [inline app/src/main/java/de/tum/in/tumcampusapp/service/FcmReceiverService.kt:31]: I dont know if this is nessesary too

File: app/src/main/java/de/tum/in/tumcampusapp/utils/Const.kt [hunk 4]
```diff
@@ -32,9 +32,8 @@ object Const {
     const val API_HOSTNAME_NEW = "app.tum.de"
     const val STUDY_ROOMS_HOSTNAME = "www.devapp.it.tum.de"
 
-    const val CURRENT_CHAT_ROOM = "current_chat_room"
+    // TODO Remove or rename
```
Comment (CommanderStorm) [inline app/src/main/java/de/tum/in/tumcampusapp/utils/Const.kt:35]: is `CHAT_ROOM_DISPLAY_NAME` still nessesary?

File: app/src/main/java/de/tum/in/tumcampusapp/utils/Utils.kt [hunk 5]
```diff
@@ -72,6 +72,22 @@ object Utils {
         return sp.getString(key, defaultVal)!!
     }
 
+    /**
+     * Get a value from the default shared preferences.
+     *
+     * TODO: This method is unused. Should we keep it?
```
Comment (CommanderStorm) [inline app/src/main/java/de/tum/in/tumcampusapp/utils/Utils.kt:78]: This will have to be decided in the review

**Comment by CommanderStorm** [comment]
@Bentipa @joschahenningsen 
Bumping this, since this is the last major pice we need for a release 😉

**Comment by Bentipa** [comment]
@CommanderStorm do you need a code review or just an approve that this removal makes sense?
Regarding a code review, I do not know if there is more stuff that could be removed as I was not involved in the initial chat integration, apart from that you can merge it from my side.
Regarding if removal makes sense, fine for me, I think it's not needed anymore :D

**Comment by CommanderStorm** [comment]
Most of the changes are probably not critical, but I have noted a few places where I was unsure how to proceed. (See my review above)
If we have an answer to the questions from my review, this should be ready to ship.
If we are still unsure after you looking at these points, we can keep it.

File: app/src/main/java/de/tum/in/tumcampusapp/utils/Utils.kt (hunk 5 above)
Comment (kordianbruck) [reply app/src/main/java/de/tum/in/tumcampusapp/utils/Utils.kt:78]: No.

File: app/src/main/java/de/tum/in/tumcampusapp/utils/Const.kt (hunk 4 above)
Comment (kordianbruck) [reply app/src/main/java/de/tum/in/tumcampusapp/utils/Const.kt:35]: No.

File: app/src/main/java/de/tum/in/tumcampusapp/service/FcmReceiverService.kt (hunk 3 above)
Comment (kordianbruck) [reply app/src/main/java/de/tum/in/tumcampusapp/service/FcmReceiverService.kt:31]: No, this is super old and can go.

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/EventsViewModel.kt (hunk 2 above)
Comment (kordianbruck) [reply app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/EventsViewModel.kt:22]: I think so still, if we want to keep the ticket system / selling

Comment (CommanderStorm) [reply app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/EventsViewModel.kt:22]:
```suggestion
```

File: app/src/main/java/de/tum/in/tumcampusapp/component/ui/onboarding/OnboardingExtrasFragment.kt [hunk 6]
```diff
@@ -100,69 +87,38 @@ class OnboardingExtrasFragment : FragmentForLoadingInBackground<ChatMember>(
...
-        // Tell the server the new member
-        val member: ChatMember?
-        try {
-            // After the user has entered their display name, create the new member on the server
-            member = tumCabeClient.createMember(currentChatMember)
-        } catch (e: IOException) {
-            Utils.log(e)
-            Utils.showToastOnUIThread(requireActivity(), R.string.error_setup_chat_member)
-            return null
-        }
```
Comment (CommanderStorm) [inline app/src/main/java/de/tum/in/tumcampusapp/component/ui/onboarding/OnboardingExtrasFragment.kt:131]: look here

**Review by joschahenningsen — APPROVED** [review]

**Pull request closed without merging**